src/AElf.OS.Core/Network/Application/ISyncStateService.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using AElf.Kernel.Blockchain.Application;
+using AElf.Kernel.Node.Application;
+using AElf.OS.Network.Infrastructure;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Volo.Abp.DependencyInjection;
+
+namespace AElf.OS.Network.Application
+{
+    public interface ISyncStateService
+    {
+        SyncState GetSyncState();
+        long GetCurrentSyncTarget();
+        Task StartSyncAsync();
+        Task UpdateSyncStateAsync();
+    }
+
+    public enum SyncState { UnInitialized, Syncing, Finished }
+
+    public class SyncStateService : ISyncStateService, ISingletonDependency
+    {
+        private NetworkOptions NetworkOptions => NetworkOptionsSnapshot.Value;
+        public IOptionsSnapshot<NetworkOptions> NetworkOptionsSnapshot { get; set; }
+        
+        private readonly INodeSyncStateProvider _syncStateProvider;
+        private readonly IBlockchainService _blockchainService;
+        private readonly IBlockchainNodeContextService _blockchainNodeContextService;
+        private readonly IPeerPool _peerPool;
+
+        public ILogger<SyncStateService> Logger { get; set; }
+        
+        public SyncStateService(INodeSyncStateProvider syncStateProvider, IBlockchainService blockchainService, 
+            IBlockchainNodeContextService blockchainNodeContextService, IPeerPool peerPool)
+        {
+            _syncStateProvider = syncStateProvider;
+            _blockchainService = blockchainService;
+            _blockchainNodeContextService = blockchainNodeContextService;
+            _peerPool = peerPool;
+        }
+        
+        public long GetCurrentSyncTarget() => _syncStateProvider.SyncTarget;
+
+        public SyncState GetSyncState()
+        {
+            switch (_syncStateProvider.SyncTarget)
+            {
+                case 0:
+                    return SyncState.UnInitialized;
+                case -1:
+                    return SyncState.Finished;
+                default:
+                    return SyncState.Syncing;
+            }
+        }
+        
+        private void SetSyncTarget(long value) => _syncStateProvider.SetSyncTarget(value);
+
+        /// <summary>
+        /// Based on current peers, will determine if a sync is needed or not. This method
+        /// should only be called once, to go from an uninitialized state to either syncing
+        /// or not syncing.
+        /// </summary>
+        /// <returns></returns>
+        public async Task StartSyncAsync()
+        {
+            if (GetSyncState() != SyncState.UnInitialized)
+            {
+                Logger.LogWarning("Trying to start the sync, but it has already been started/finished.");
+                return;
+            }
+
+            await TryFindSyncTargetAsync();
```

Comment:
if(await TryFindSyncTargetAsync())....

or do not name it Try.....

Reply:
ah yes, my intention was not to make it a "Try..." method, renamed.